UPtables/spec/controllers/ects_modules_controller_spec.rb

```diff
@@ -0,0 +1,161 @@
+
+RSpec.describe EctsModulesController, type: :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # EctsModule. As you add validations to EctsModule, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    {:name => "I am a module"}
+  }
+
+  let(:invalid_attributes) {
+    {:name => nil}
+  }
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # EctsModulesController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all ects_modules as @ects_modules" do
+      ects_module = EctsModule.create! valid_attributes
+      get :index, {}, valid_session
+      expect(assigns(:ects_modules)).to eq([ects_module])
+    end
+  end
+
+  describe "GET #show" do
+    it "assigns the requested ects_module as @ects_module" do
+      ects_module = EctsModule.create! valid_attributes
+      get :show, {:id => ects_module.to_param}, valid_session
+      expect(assigns(:ects_module)).to eq(ects_module)
+    end
+  end
+
+  describe "GET #new" do
+    it "assigns a new ects_module as @ects_module" do
+      get :new, {}, valid_session
+      expect(assigns(:ects_module)).to be_a_new(EctsModule)
+    end
+  end
+
+  describe "GET #edit" do
+    it "assigns the requested ects_module as @ects_module" do
+      ects_module = EctsModule.create! valid_attributes
+      get :edit, {:id => ects_module.to_param}, valid_session
+      expect(assigns(:ects_module)).to eq(ects_module)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new EctsModule" do
+        expect {
+          post :create, {:ects_module => valid_attributes}, valid_session
+        }.to change(EctsModule, :count).by(1)
+      end
+
+      it "assigns a newly created ects_module as @ects_module" do
+        post :create, {:ects_module => valid_attributes}, valid_session
+        expect(assigns(:ects_module)).to be_a(EctsModule)
+        expect(assigns(:ects_module)).to be_persisted
+      end
+
+      it "redirects to the created ects_module" do
+        post :create, {:ects_module => valid_attributes}, valid_session
+        expect(response).to redirect_to(EctsModule.last)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved ects_module as @ects_module" do
+        post :create, {:ects_module => invalid_attributes}, valid_session
+        expect(assigns(:ects_module)).to be_a_new(EctsModule)
+      end
+
+      it "re-renders the 'new' template" do
+        post :create, {:ects_module => invalid_attributes}, valid_session
+        expect(response).to render_template("new")
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "with valid params" do
+      let(:new_attributes) {
+        {:name => "Module with some courses", :course_ids => [23, 42]}
```

Comment:
@felixKubicek this is the corresponding controller spec, which will fail if we don't permit the ```:course_ids```